documentation/src/main/asciidoc/manual-index.asciidoc

```diff
@@ -361,3 +361,226 @@ parameters through - see <<lucene-indexing-performance>> for more details.
 The MassIndexer uses a forward only scrollable result to iterate on the primary keys to be loaded,
 but MySQL's JDBC driver will load all values in memory; to avoid this "optimization" set
 `idFetchSize` to `Integer.MIN_VALUE`.
+
+
+[[jsr352-integration]]
+=== Integration with JSR-352
+
+[CAUTION]
+====
+This feature is a work in progress.
+====
+
+The integration with JSR-352, Batch Applications for the Java Platform, is in development. We do
+think we have the basics covered and we are looking for feedback.
+
+Hibernate Search provides a new mass indexing approach using batch job introduced by JSR352.
+It covers not only the existing functionality of the previous indexer, but also benefits the
+powerful standard based facilities of the Java Batch Platform (JSR-352), such as checkpoint and
+step partitioning. However, it requires a batch runtime for the execution:
+
+* For Java EE users, we use the batch runtime provided by your Java EE container.
+* For Java SE users, we use the batch runtime
+  https://github.com/jberet/jsr352/tree/master/jberet-se[JBeret SE].
+
+.How to use the new mass-indexer?
+====
+[source, JAVA]
+----
+Properties jobProps = MassIndexingJob
+        .parameters()
+        .forEntity( MyEntity.class )
+        .build();
+long executionId = BatchRuntime
+        .getJobOperator()
+        .start( MassIndexingJob.NAME, jobProps );
+----
+====
+
+In the following paragraphs, we will explain why JSR-352 standard is a good choice, and how to
+configure different parameters in order to bootstrap the mass-indexing job.
+
+==== Define your indexing scope
+Our job allows you to define your own indexing scope—you can start a full indexation or a partial
+indexation through 3 different methods: selecting the desired entity types, using HQL, or using
+Hibernate criteria.
```

Comment:
The _HQL_ method does not support partitioning, but the _Criteria_ method does support partitioning. The key point is whether the selection, given by user, is orderable.